cmake_test/add_test.cmake

```diff
@@ -1,5 +1,20 @@
 include_guard()
-include(cmake_test/detail_/add_test)
+include(cmake_test/detail_/utilities/print_result)
```

Comment:
Can we break this file up one function per file? Also these functions need documented and unit tested (to the extent possible since it's going to be circular to unit test CMakeTest with CMakeTest...).